ceph_salt/validate/config.py

```diff
@@ -18,13 +19,18 @@ def validate_config(host_ls):
             return "No bootstrap Mon IP specified in config"
         if bootstrap_mon_ip in ['127.0.0.1', '::1']:
             return 'Mon IP cannot be the loopback interface IP'
-    time_server_host = PillarManager.get('ceph-salt:time_server:server_host')
-    if not time_server_host:
-        return 'No time server host specified in config'
-    time_server_subnet = PillarManager.get('ceph-salt:time_server:subnet')
-    if not time_server_subnet:
-        return 'No time server subnet specified in config'
-    all_minions = PillarManager.get('ceph-salt:minions:all')
+    time_server_enabled = PillarManager.get('ceph-salt:time_server:enabled', True)
+    if time_server_enabled:
+        time_server_host = PillarManager.get('ceph-salt:time_server:server_host')
+        if not time_server_host:
+            return 'No time server host specified in config'
+        time_server_is_minion = time_server_host in all_minions
+        time_server_subnet = PillarManager.get('ceph-salt:time_server:subnet')
+        if time_server_is_minion and not time_server_subnet:
+            return 'No time server subnet specified in config'
```

Comment:
Can we also validate that if `time_server_is_minion`, then `time_server_subnet` should not be specified?